internal/splunk/httprequest.go

```diff
@@ -53,7 +55,12 @@ func HandleHTTPCode(resp *http.Response) error {
 		err = exporterhelper.NewThrottleRetry(err, time.Duration(retryAfter)*time.Second)
 	// Check for permanent errors.
 	case http.StatusBadRequest, http.StatusUnauthorized:
-		err = consumererror.Permanent(err)
+		var dump []byte
+		dump, err = httputil.DumpResponse(resp, true)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
log.error? Also consider to pass the `logger` from the components instead of using the golang log.

Reply:
Please don't log here, especially not Fatal which exits the program. Instead just return an error. If logging is absolutely needed for debugging use logger.Debug (and use the zap logger as Bogdan recommends). It seems returning an error is sufficient here.

See sections here  https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#coding-guidelines:
>Do not Crash after Startup
>Logging

Reply:
Ok, we are now returning the error instead